src/voxelisers/bvh-ray-thick.ts

```diff
@@ -0,0 +1,140 @@
+            const positionData = mesh.getVertices(triIndex);
+            triangles[triIndex] = [positionData.v0, positionData.v1, positionData.v2];
+        }
+
+        const MAX_TRIANGLES_PER_NODE = 8;
+        LOG('Creating BVH...');
+        const bvh = new bvhtree.BVH(triangles, MAX_TRIANGLES_PER_NODE);
+        LOG('BVH created...');
+
+        // Generate rays
+        const bounds = mesh.getBounds();
+        bounds.min.floor();
+        bounds.max.ceil();
+
+        const planeDims = Vector3.sub(bounds.max, bounds.min).add(1);
+        const numRays = (planeDims.x * planeDims.y) + (planeDims.x * planeDims.z) + (planeDims.y * planeDims.z);
+        const rays = new Array<Ray>(numRays);
+        let rayIndex = 0;
+        {
+            // Generate x-plane rays
+            for (let y = bounds.min.y; y <= bounds.max.y; ++y) {
+                for (let z = bounds.min.z; z <= bounds.max.z; ++z) {
+                    rays[rayIndex++] = {
+                        origin: new Vector3(bounds.min.x - 1, y, z),
+                        axis: Axes.x,
+                    };
+                }
+            }
+            // Generate y-plane rays
+            for (let x = bounds.min.x; x <= bounds.max.x; ++x) {
+                for (let z = bounds.min.z; z <= bounds.max.z; ++z) {
+                    rays[rayIndex++] = {
+                        origin: new Vector3(x, bounds.min.y - 1, z),
+                        axis: Axes.y,
+                    };
+                }
+            }
+            // Generate z-plane rays
+            for (let x = bounds.min.x; x <= bounds.max.x; ++x) {
+                for (let y = bounds.min.y; y <= bounds.max.y; ++y) {
+                    rays[rayIndex++] = {
+                        origin: new Vector3(x, y, bounds.min.z - 1),
+                        axis: Axes.z,
+                    };
+                }
+            }
+        }
+        ASSERT(rays.length === rayIndex);
+        LOG('Rays created...');
+
+        const VecZero = new Vector3(0, 0, 0)
+
+        // Ray test BVH
+        const taskHandle = ProgressManager.Get.start('Voxelising');
+        for (rayIndex = 0; rayIndex < rays.length; ++rayIndex) {
+            ProgressManager.Get.progress(taskHandle, rayIndex / rays.length);
+
+            const ray = rays[rayIndex];
+            const intersections = bvh.intersectRay(ray.origin, axesToDirection(ray.axis), false);
+            for (const intersection of intersections) {
+                const point = intersection.intersectionPoint;
+                const position = new Vector3(point.x, point.y, point.z);
+                
+                // // Shrinking towards the center
+                // const centerVector = VecZero.copy().sub(position)
+                // const depthPosition = position.copy().add(centerVector.normalise()).round()
+                
+                // Shrinking towards the perpendicular vector
+                const triangle = intersection.triangle
```

Comment:
Should be able to use `const triangle = mesh.getUVTriangle(intersection.triangleIndex);` and the `p0` will be `triangle.v0`, `p1` is `triangle.v1` and `p2` is `triangle.v2`.